flink-connector-kafka/src/main/java/org/apache/flink/connector/kafka/source/enumerator/KafkaSourceEnumerator.java

```diff
@@ -113,10 +124,13 @@ public KafkaSourceEnumerator(
             Properties properties,
             SplitEnumeratorContext<KafkaPartitionSplit> context,
             Boundedness boundedness,
-            Set<TopicPartition> assignedPartitions) {
+            Set<TopicPartition> assignedPartitions,
```

Comment:
It will be better to pass the `KafkaSourceEnumState` object in this constructor to limit the number of arguments.